### PR TITLE
add media player via room api

### PR DESCRIPTION
**Pull request by nalin29**
Adds Room based audio. Need to fall back on data coordinator as Room does not support websocket. But we can just create 2 entities for this an old coordinator one used by rooms and websocket for the rest

### CODE REVIEW

**Comment by nalin29** [comment]
I did digging on this when I tried to implement first with websockets. There is no actual room object to subscribe to for events. The way it seems to work is that it will get various updates across multiple devices. So the device for media player will contain an update that it is playing in that room. Then there might be a separate device like the audio matrix mentioning the stats for the different connections like audio and input now. Its just a mess and quite hard to parse. I don't know if the app has some hidden api that makes this easier. The data coordinator is significantly simpler as we can treat each room as an object and query its state, instead of trying to piece it together.

**Comment by lawtancool** [comment]
Hey @nalin29, I added some features to your branch and merged the PR, let me know how it works on your system!

- Added HA grouping support by setting group entities to same C4 RoomSource
- Added media image, artist, album, etc. info
- Added device area and extra state attributes to CoordinatorEntity
- Added scan interval to options flow
- Only set up room entities that have valid experiences (listen, video)
